src/Algebra/Properties/Semiring/Exp.agda

```diff
@@ -34,10 +34,37 @@ open import Algebra.Definitions.RawSemiring rawSemiring public
 ^-cong : _^_ Preserves₂ _≈_ ⟶ _≡_ ⟶ _≈_
 ^-cong x≈y u≡v = Mult.×-cong u≡v x≈y
 
+^-congʳ : ∀ x → (x ^_) Preserves _≡_ ⟶ _≈_
+^-congʳ x = Mult.×-congˡ
+
 -- xᵐ⁺ⁿ ≈ xᵐxⁿ
 ^-homo-* : ∀ x m n → x ^ (m ℕ.+ n) ≈ (x ^ m) * (x ^ n)
 ^-homo-* = Mult.×-homo-+
 
 -- (xᵐ)ⁿ≈xᵐ*ⁿ
 ^-assocʳ : ∀ x m n → (x ^ m) ^ n ≈ x ^ (m ℕ.* n)
 ^-assocʳ x m n rewrite ℕ.*-comm m n = Mult.×-assocˡ x n m
+
+------------------------------------------------------------------------
+-- A lemma using commutativity, needed for the Binomial Theorem
+
+module _ {x} {y} (x*y≈y*x : x * y ≈ y * x) where
```

Comment:
I think it's best just to stick these parameters in the type rather than having a module. We usually use modules to generalise over assumptions about operators that apply to multiple proofs. It's slightly weird to have one which refers to properties of specific elements.

Speaking of which is there a reason why you're not simply assuming `_*_` is `Commutative`?

Reply:
Regarding explicit `module` parametrisation over putting them in the type; these were 'ambient' assumptions across a number of parts of the development. But for sure that's not a deal-breaker. 

Regarding the assumption itself, for sure it means that the lemmas (and the eventual theorem) are an easy corollary for `CommutativeSemiring`. But I was reasonably persuaded by the [Wikipedia article](https://en.wikipedia.org/wiki/Binomial_theorem#In_abstract_algebra) of the greater utility in having it stated for individual `x`, `y` (in a possibly non-commutative (semi)ring, eg of matrices) which happen to commute, than insistign on commutativity up-front.

Reply:
Fair enough :+1: